task_manager/task_manager.cc

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "task_manager.hh"
+
+logging::logger tmlogger("task manager");
+
+future<> task_manager::stop() noexcept {
+    if (!_modules.empty()) {
+        on_internal_error(tmlogger, "Tried to stop task manager while some modules were not unregistered");
+    }
+    return make_ready_future<>();
+}
+
+future<task_manager::foreign_task_ptr> task_manager::lookup_task_on_all_shards(sharded<task_manager>& tm, task_id tid) {
+    task_manager::foreign_task_ptr res;
+    co_await parallel_for_each(boost::irange(0u, smp::count), [&tm, tid, &res] (unsigned shard) -> future<> {
```

Comment:
nit: use coroutine::parallel_for_each (it saves an allocation, not that we care so much about performance on this path, but it's a good habit)